src/estimators/stratified_sketch.py

```diff
@@ -0,0 +1,261 @@
+    self.sketch_intersection = sketch_operator.intersection
+
+  def __call__(self, this, that):
+    merged = self.merge_sketches(this, that)
+    return merged
+
+  def merge_sketches(self, this, that):
+    """Merge two StratifiedSketch.
+
+       Given 2 sketches A and B:
+       Merged(k) = (A(k) & B(0)) U (A(k-1) & B(1)) ... U (A(0) & B(k))
+         where
+         A(k) & B(0) =  A(k) - (A(k) & B(1+))
+         B(k) & A(0) =  B(k) - (B(k) & A(1+))
+    Args:
+      this: one of the two StratifiedSketch to be merged.
+      that: the other StratifiedSketch to be merged.
+
+    Returns:
+      A merged StratifiedSketch from the input.
+    """
+    assert isinstance(this, StratifiedSketch)
+    this.assert_compatible(that)
+    max_freq = this.max_freq
+    merged_sketch = copy.deepcopy(this)
+
+    for k in range(1, max_freq):
+      # Calculate A(k) & B(0) = A(k) - (A(k) & B(1+))
+      merged = self.sketch_difference(
+          this.sketches[k],
+          self.sketch_intersection(this.sketches[k], that.sketches[ONE_PLUS]))
+
+      # Calculate A(0) & B(k) = B(k) - (B(k) & A(1+))
+      merged = self.sketch_union(
+          merged,
+          self.sketch_difference(
+              that.sketches[k],
+              self.sketch_intersection(this.sketches[ONE_PLUS],
+                                       that.sketches[k])))
+
+      # Calculate A(i) & B(k-i)
+      for i in range(1, k):
+        merged = self.sketch_union(
+            merged,
+            self.sketch_intersection(this.sketches[i], that.sketches[(k - i)]))
+      merged_sketch.sketches[k] = merged
+
+    # Calculate Merged(max_freq)
+    merged = this.sketches[max_freq]
+    rest = that.sketches[ONE_PLUS]
+    for k in range(1, max_freq):
+      merged = self.sketch_union(
+          merged, self.sketch_intersection(this.sketches[max_freq - k], rest))
+      rest = self.sketch_difference(rest, that.sketches[k])
+    merged = self.sketch_union(
+        merged,
+        self.sketch_difference(
+            that.sketches[max_freq],
+            self.sketch_intersection(that.sketches[max_freq],
+                                     this.sketches[ONE_PLUS])))
+    merged_sketch.sketches[max_freq] = merged
+
+    # Calculate Merged(1+)
+    merged_one_plus = None
+    for k in range(max_freq):
+      merged_one_plus = self.sketch_union(merged_one_plus,
+                                          merged_sketch.sketches[k + 1])
+    merged_sketch.sketches[ONE_PLUS] = merged_one_plus
+
+    return merged_sketch
+
+
+class SequentialEstimator(object):
+  """Sequential frequency estimator."""
+
+  def __init__(self, sketch_operator):
+    self.pairwise_estimator = PairwiseEstimator(sketch_operator)
+
+  def __call__(self, sketches_list):
+    return self.merge_sketches(sketches_list)
+
+  def merge_sketches(self, sketches_list):
+    return functools.reduce(self.pairwise_estimator.merge_sketches,
+                            sketches_list)
+
+
+class ExactMultiSetOperation(object):
```

Comment:
Should this be called ExactMultiSetOperation?  Perhaps a better name would be ExactSetOperation, since the frequencies are being set to one?

Reply:
yes, I was thinking about that too. I was just not sure since we don't have an ExactSet class. 
Changed.